lib/jira/client/index.js

```diff
@@ -152,28 +142,27 @@ module.exports = async (jiraHost, gitHubInstallationId, logger) => {
  * Splits commits in data payload into chunks of 400 and makes separate requests
  * to avoid Jira API limit
  */
-const batchedCommitRequest = async (data, options, instance, installationId) => {
-  const filteredCommits = dedupCommits(data)
+const batchedBulkUpdate = async (data, options, instance, installationId) => {
+  const dedupedCommits = data.commits ? dedupCommits(data.commits) : []
 
-  const commitChunks = []
-  while (filteredCommits.length) {
-    commitChunks.push(filteredCommits.splice(0, 400))
+  // Initialize with an empty chunk of commits so we still process the request if there are no commits in the payload
+  const commitChunks = [[]]
+  while (dedupedCommits.length) {
+    commitChunks.push(dedupedCommits.splice(0, 400))
   }
 
-  const batchedUpdates = []
-  commitChunks.forEach((commitChunk) => {
-    const repositoryObj = {
-      ...data,
-      commits: commitChunk
+  const batchedUpdates = commitChunks.map((commitChunk) => {
+    if (commitChunk.length) {
+      data.commits = commitChunk
     }
 
-    batchedUpdates.push(instance.post('/rest/devinfo/0.10/bulk', {
+    return instance.post('/rest/devinfo/0.10/bulk', {
       preventTransitions: (options && options.preventTransitions) || false,
-      repositories: [repositoryObj],
+      repositories: [data],
```

Comment:
In it's current state, we'll resend any data outside of the commits when we exceed 400 commits. That raises two questions in my mind:

1. Is it problematic to send certain data twice? This one we need to answer before shipping this. I suspect it isn't a problem but it's worth checking.
2. Is there some data we can remove from subsequent requests? For example, on subsequent requests, can we drop `pullRequests` and `branches`, if present? This one we can punt to a later PR.

Reply:
The description for the field in the docs is:

> List of commits to update in this repository. Must not contain duplicate entity IDs. Maximum number of commits is 400

I'm reasonably confident that we won't run into any issues since it only "updates" the list of commits when sending multiple requests as long as we use the same repository `id` for each request. I think it's a good idea to drop `pullRequests` and `branches` in a follow up (possibly in addition to adding support for batching other entities)